src/metrics.jl

```diff
@@ -720,33 +721,34 @@ function _pairwise!(r::AbstractMatrix, dist::Euclidean,
                     a::AbstractMatrix, b::AbstractMatrix)
     m, na, nb = get_pairwise_dims(r, a, b)
     mul!(r, a', b)
-    sa2 = sumsq_percol(a)
-    sb2 = sumsq_percol(b)
+    sa2 = sum(abs2, a, dims=1)
+    sb2 = sum(abs2, b, dims=1)
     threshT = convert(eltype(r), dist.thresh)
     if threshT <= 0
+        # If there's no chance of triggering the threshold, we can use @simd
         for j = 1:nb
             sb = sb2[j]
             @simd for i = 1:na
                  @inbounds r[i, j] = sqrt(max(sa2[i] + sb - 2 * r[i, j], 0))
             end
         end
     else
-        @inbounds for j = 1:nb
```

Comment:
Why change this?

Reply:
We had slightly different implementations for `SqEuclidean` and `Euclidean`, and the former was faster. Now that I think about it again, I'll reduce the code duplication using `eval_end`.